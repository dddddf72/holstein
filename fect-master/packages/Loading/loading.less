.fect-loading-container {
  display: inline-block;
  align-items: center;
  position: relative;
  width: 100%;
  height: 100%;
  .loading {
    position: absolute;
    top: 0;
    // left: 0;
    right: 0;
    bottom: 0;
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: transparent;
    user-select: none;
  }
  i {
    width: 5px;
    height: 5px;
    border-radius: 50%;
    margin: 0 1px;
    display: inline-block;
    animation: loading-blink 1.4s infinite both;
    background-color: var(--accents-7);
    &:nth-child(2) {
      animation-delay: 0.2s;
    }
    &:nth-child(3) {
      animation-delay: 0.4s;
    }
  }
  @keyframes loading-blink {
    0% {
      opacity: 0.2;
    }

    20% {
      opacity: 1;
    }

    100% {
      opacity: 0.2;
    }
  }

  i {
    &.mini {
      width: 2px;
      height: 2px;
    }
    &.small {
      height: 3px;
      width: 3px;
    }
    &.large {
      width: 7px;
      height: 7px;
    }
    &.success {
      background-color: var(--success-default);
    }
    &.warning {
      background-color: var(--warning-default);
    }
    &.error {
      background-color: var(--error-dark);
    }
  }
}
