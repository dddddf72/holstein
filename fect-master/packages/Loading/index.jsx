import { toRefs } from 'vue'
import { createNameSpace, validator, theme } from '../utils'

const { normalSizes, normalTypes } = theme

const [createComponent] = createNameSpace('Loading')
import './loading.less'

export default createComponent({
  props: {
    size: {
      type: String,
      validator: validator.enums(normalSizes),
      default: 'medium',
    },
    type: {
      type: String,
      validator: validator.enums(normalTypes),
      default: 'default',
    },
    color: {
      type: String,
      default: '',
    },
  },
  setup(props, { attrs }) {
    const { size, type, color } = toRefs(props)
    const safeColor = !!color?.value
    return () => (
      <div
        {...attrs}
        className={`fect-loading-container ${attrs.class ? attrs.class : ''}`}
      >
        <span className={'loading'}>
          {new Array(3).fill(0).map((item, i) => (
            <i
              className={`${size.value} ${type.value}`}
              style={{ backgroundColor: `${safeColor && color.value}` }}
              key={item + i}
            ></i>
          ))}
        </span>
      </div>
    )
  },
})
