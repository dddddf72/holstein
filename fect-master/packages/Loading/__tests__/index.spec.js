import { mount } from '@vue/test-utils'
import { ref } from 'vue'
import Loading from '../index'

describe('Loading', () => {
  it('should be render as element', () => {
    const loading = mount(Loading)
    expect(loading.html()).toMatchSnapshot()
    expect(() => loading.unmount()).not.toThrow()
  })

  it('should be support types', () => {
    const wrapper = mount({
      setup() {
        return { typeList: ['default', 'success', 'warning', 'error'] }
      },
      render(h) {
        return (
          <>
            {this.typeList.map((item, i) => (
              <Loading type={item} key={i} />
            ))}
          </>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('.fect-loading-container').length).toEqual(4)
  })

  it('should be support different sizes', () => {
    const wrapper = mount({
      setup() {
        return { sizeList: ['mini', 'small', 'medium', 'large'] }
      },
      render(h) {
        return (
          <>
            {this.sizeList.map((item, i) => (
              <Loading size={item} key={i} />
            ))}
          </>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('.fect-loading-container').length).toEqual(4)
  })

  it('should be support use custom color', () => {
    const wrapper = mount({
      setup() {
        const cusColor = ref('pink')
        return { cusColor }
      },
      render() {
        return (
          <>
            <Loading color={this.cusColor} />
          </>
        )
      },
    })
    const el = wrapper.find('i')
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.vm.cusColor).toEqual('pink')
    expect(el.attributes('style')).toBe('background-color: pink;')
  })
})
