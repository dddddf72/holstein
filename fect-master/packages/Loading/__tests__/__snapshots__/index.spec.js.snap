// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should be render as element 1`] = `<div class="fect-loading-container "><span class="loading"><i class="medium default"></i><i class="medium default"></i><i class="medium default"></i></span></div>`;

exports[`Loading should be support different sizes 1`] = `
<div class="fect-loading-container "><span class="loading"><i class="mini default"></i><i class="mini default"></i><i class="mini default"></i></span></div>
<div class="fect-loading-container "><span class="loading"><i class="small default"></i><i class="small default"></i><i class="small default"></i></span></div>
<div class="fect-loading-container "><span class="loading"><i class="medium default"></i><i class="medium default"></i><i class="medium default"></i></span></div>
<div class="fect-loading-container "><span class="loading"><i class="large default"></i><i class="large default"></i><i class="large default"></i></span></div>
`;

exports[`Loading should be support types 1`] = `
<div class="fect-loading-container "><span class="loading"><i class="medium default"></i><i class="medium default"></i><i class="medium default"></i></span></div>
<div class="fect-loading-container "><span class="loading"><i class="medium success"></i><i class="medium success"></i><i class="medium success"></i></span></div>
<div class="fect-loading-container "><span class="loading"><i class="medium warning"></i><i class="medium warning"></i><i class="medium warning"></i></span></div>
<div class="fect-loading-container "><span class="loading"><i class="medium error"></i><i class="medium error"></i><i class="medium error"></i></span></div>
`;

exports[`Loading should be support use custom color 1`] = `<div class="fect-loading-container "><span class="loading"><i class="medium default" style="background-color: pink;"></i><i class="medium default" style="background-color: pink;"></i><i class="medium default" style="background-color: pink;"></i></span></div>`;
